File: hackerrank/30Days/Day8.java
/*Given  names and phone numbers, assemble a phone book that maps friends' names to their respective phone numbers. You will then be given an unknown number of names to query your phone book for. For each queried, print the associated entry from your phone book on a new line in the form name=phoneNumber; if an entry for  is not found, print Not found instead.*/
import java.util.*;
import java.io.*;

class Solution{
    public static void main(String []argh){
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        HashMap<String,Integer> dict = new HashMap<String,Integer>();
        for(int i = 0; i < n; i++){
            String name = in.next();
            int phone = in.nextInt();
            dict.put(name,(Integer)phone);
            
        }
        while(in.hasNext()){
            String s = in.next();
            if(dict.containsKey(s)){
                System.out.println(s + "=" + dict.get(s));
            }
            else{
                System.out.println("Not found");
            }
        }
        in.close();
    }
}

File: hackerrank/30Days/Day8.py
#Given  names and phone numbers, assemble a phone book that maps friends' names to their respective phone numbers. You will then be given an unknown number of names to query your phone book for. For each queried, print the associated entry from your phone book on a new line in the form name=phoneNumber; if an entry for  is not found, print Not found instead.

myDict = {}
t = int(raw_input())
p = t
while t:
    s = raw_input()
    l = s.split(' ')
    myDict[l[0]] = l[1]
    t-=1
while p:
    name = raw_input()
    try:
        q = name
        q += "={0}".format(myDict[name])
    except:
        q = "Not found"
    print q
    p-=1
